### PR TITLE
[ISSUE-407] Improve e2e testing due to issues found with CI

### PR DESCRIPTION
**Pull request by safronovD**
### Issue #407 

1. CSI Operator deploys in each test case in the same namespace now (logs collection works correctly)
2. Test case fails if kube-scheduler is not restarted after patching
3. Increase waiting time in node-label test
4. Update `GetNodePodsNames`, now it works with specific daemonset (like csi-baremetal-node-kernel-5.4)

## PR checklist
- [x] Add link to the issue
- [x] Choose PR label
- [ ] New unit tests added
- [ ] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved

## Testing
_Link to custom CI build_

### CODE REVIEW

File: test/e2e/common/deploy.go [hunk 1]
```diff
@@ -110,13 +113,22 @@ func DeployCSI(f *framework.Framework, additionalInstallArgs string) (func(), er
...
+		err             error
 	)
 
 	if additionalInstallArgs != "" {
 		installArgs += " " + additionalInstallArgs
 	}
 
 	cleanup := func() {
+		// Collect and print Operator logs
+		if logs, err := GetOperatorLogs(f.ClientSet); err != nil {
```
Comment (mishoyama) [inline test/e2e/common/deploy.go:126]: Can we see operator logs in runtime?

**Review by mishoyama — CHANGES_REQUESTED** [review]

File: test/e2e/scenarios/node-labels.go [hunk 2]
```diff
@@ -82,7 +82,7 @@ func labeledDeployTestSuite() {
 		// wait till operator reconcile csi
 		// operator has to receive NodeUpdate request and label new nodes to expand node-daemonset
 		// if new pods aren't created in time, WaitForPodsRunningReady will be skipped
-		deadline := time.Now().Add(time.Second * 30)
+		deadline := time.Now().Add(2 * time.Minute)
```
Comment (painhardcore) [inline test/e2e/scenarios/node-labels.go:85]: Why time increased from 30 seconds up to 2 minutes?

File: test/e2e/common/deploy.go [hunk 3]
```diff
@@ -151,13 +163,18 @@ func DeployCSI(f *framework.Framework, additionalInstallArgs string) (func(), er
...
 	if schedulerRC.IsInitialized {
-		if isRestarted, err := schedulerRC.WaitForRestart(); err != nil {
+		if isRestarted, err = schedulerRC.WaitForRestart(); err != nil {
 			e2elog.Logf("SchedulerRestartChecker has been failed while waiting. Err: %s", err)
-		} else {
-			e2elog.Logf("Scheduler is restarted: %t", isRestarted)
 		}
 	}
 
+	if isRestarted {
```
[the remark this answers is not in the document]
Comment (safronovD) [reply test/e2e/common/deploy.go:187]: updated the heading, description and issue

File: test/e2e/common/deploy.go (hunk 1 above)
Comment (safronovD) [reply test/e2e/common/deploy.go]: reworked, CSI Operator deploys in the same namespace

File: test/e2e/scenarios/node-labels.go (hunk 2 above)
Comment (safronovD) [reply test/e2e/scenarios/node-labels.go:80]: We have some fails in CI when CSI Operator has no time to label nodes. If it works correctly, the loop will be stopped after success, if no, we will have more info about a failure

**Review by mishoyama — APPROVED** [review]

**Review by painhardcore — APPROVED** [review]

**Review by yurkov-anton — APPROVED** [review]

**Pull request merged**